### PR TITLE
Removes `as [disambiguation_path]` from `derive_impl` usage

### PR DESCRIPTION
**Pull request by gupnik**
Step in https://github.com/paritytech/polkadot-sdk/issues/171

This PR removes `as [disambiguation_path]` syntax from `derive_impl` usage across the polkadot-sdk as introduced in https://github.com/paritytech/polkadot-sdk/pull/3505

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5497458

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5497715

File: substrate/frame/support/src/tests/mod.rs [hunk 1]
```diff
@@ -227,7 +227,7 @@ crate::construct_runtime!(
 	}
 );
 
-#[crate::derive_impl(self::frame_system::config_preludes::TestDefaultConfig as frame_system::DefaultConfig)]
+#[crate::derive_impl(self::frame_system::config_preludes::TestDefaultConfig as self::frame_system::DefaultConfig)]
```
Comment (liamaharon) [inline substrate/frame/support/src/tests/mod.rs:230]: Still need `as` here?

**Review by liamaharon — APPROVED** [review]

File: substrate/frame/support/src/tests/mod.rs (hunk 1 above)
Comment (gupnik) [reply substrate/frame/support/src/tests/mod.rs:230]: Yeah, as the trait is scoped under `self::frame_system` and not `self` as expected by the macro for default behaviour.

**Review by ggwpez — APPROVED** [review]

**Review by bkontur — APPROVED** [review]

**Pull request merged**